elytron/WFCORE-4362-configurable-certificate-authority-in-certificate-authority-account.adoc

```diff
@@ -0,0 +1,94 @@
+This issue aims to make it possible for the user to configure attribute `certificate-authority` for this resource.
+Each certificate authority can be associated with a production url and a staging url, both of which have to be configurable.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFCORE-4362[WFCORE-4362]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1150[EAP7-1150]
+
+=== Dev Contacts
+
+* mailto:dvilkola@redhat.com[Diana Vilkolakova]
+
+=== QE Contacts
+
+* mailto:mchoma@redhat.com[Martin Choma]
+
+=== Testing By
+[x] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+* Wildfly Core
+* Elytron
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+It must be possible to specify staging url and production url for the attribute `certificate-authority` in the `certificate-authority-account` resource.
+This will be achieved by adding new resource `certificate-authority` to the elytron subsystem.
+
+Eg. the configuration will look like the following:
+
+```
+/subsystem=elytron/certificate-authority=exampleCertificateAuthority:add(name=Buypass,production-url="https://api.buypass.com/acme/directory",staging-url="https://api.staging-example.com/acme/directory")
+/subsystem=elytron/certificate-authority-account=exampleCertificateAuthorityAccount:add(name=Buypass,alias=example,key-store=exampleKS)
+```
+
+== Implementation Plan
+
+Currently, `Certificate-authority-account` contains optional attribute `certificate-authority` of type STRING.
+The default and only allowed value is "LetsEncrypt".
+We should be able to specify staging URL and production URL without breaking backwards compatibility.
+
+In the Elytron subsystem, LetsEncrypt is a value of CertificateAuthority enum. This enum contains name, url and staging url.
+We can add new `certificate-authority` resource to the elytron
+subsystem with non-optional attributes `name`, `production-url` and optional attribute `staging-url`.
+This way, during the configuration of `certificate-authority-account`
+it will be possible to reference any `certificate-authority` previously added by attribute `name`.
+
+
+The default value for `certificate-authority` attribute in `certificate-authority-account` will continue to be "LetsEncrypt" and will have the following specified:
+
+```
+<certificate-authorities>
+    <certificate-authority name="LetsEncrypt" production-url="https://acme-v02.api.letsencrypt.org/directory" staging-url="https://acme-staging-v02.api.letsencrypt.org/directory" />
+</certificate-authorities>
+```
+
+Implementation will require removal of the enum CertificateAuthority and adding new class CertificateAuthorityDefinition that extends SimpleResourceDefinition.
+Since AcmeAccountService class (service responsible for a single AcmeAccount instance) uses CertificateAuthority enum,
+it will be changed to use certificate authority class instead.
+
+== Test Plan
+
+Functionality tests added to Elytron testsuite.
```

Comment:
Just a minor clarification, the functionality tests will be added to the Elytron tests in the Core testsuite, right?

Reply:
@fjuma yes, I specified tests plan more.